### PR TITLE
Request can be created and sent via a message with an expiration date that has passed

### PR DESCRIPTION
**Pull request by sebbi08**
# Readiness checklist

- [x] I added/updated tests.
- [x] I ensured that the PR title is good enough for the changelog.
- [x] I labeled the PR.
- [x] I self-reviewed the PR.

### CODE REVIEW

File: packages/consumption/src/consumption/ConsumptionCoreErrors.ts
Comment (jkoenig134) [inline packages/consumption/src/consumption/ConsumptionCoreErrors.ts:334]: newline

File: packages/consumption/test/modules/requests/OutgoingRequestsController.test.ts
Comment (jkoenig134) [inline packages/consumption/test/modules/requests/OutgoingRequestsController.test.ts:228]:
```suggestion
        test("returns a validation result that contains an error for requests that are expired", async function () {
```

File: packages/consumption/src/consumption/ConsumptionCoreErrors.ts
Comment (sebbi08) [reply packages/consumption/src/consumption/ConsumptionCoreErrors.ts:334]: woops already did that but forgot to push

**Review by jkoenig134 — CHANGES_REQUESTED** [review]
one small thing

**Review by jkoenig134 — APPROVED** [review]
LGTM

File: packages/consumption/src/consumption/ConsumptionCoreErrors.ts
Comment (Milena-Czierlinski) [inline packages/consumption/src/consumption/ConsumptionCoreErrors.ts:336]:
```suggestion
        return new CoreError("error.consumption.requests.cannotCreateRequestWithExpirationDateInPast", "You cannot create a Request with an expiration date that is in the past.");
```

File: packages/runtime/src/useCases/common/RuntimeErrors.ts
Comment (Milena-Czierlinski) [inline packages/runtime/src/useCases/common/RuntimeErrors.ts:134]:
```suggestion
            "The Message cannot be sent as the containing Request is already expired. Please create a new Request and try again."
```

Comment (jkoenig134) [reply packages/runtime/src/useCases/common/RuntimeErrors.ts:134]: contained would be better wording, right?

**Comment by jkoenig134** [comment]
ehm, I removed this from draft b/c I thought it's a mistake, but it looks like you set this manually

File: packages/consumption/test/modules/requests/RequestsIntegrationTest.ts
Comment (Milena-Czierlinski) [inline packages/consumption/test/modules/requests/RequestsIntegrationTest.ts:331]: In my opinion it is much easier to remove this here and simply add a sleep to those tests that require an expiration time to pass. Looking only at the tests, I didn't understand why we don't have a sleep there, since the true that was passed to `anOutgoingRequestWith` doesn't make it obvious, what's going on. Also, you don't have to adjust `ICoreDate` then. Exposing a DateTime there doesn't feel like something we want to do.

File: packages/runtime/test/transport/messages.test.ts
Comment (Milena-Czierlinski) [inline packages/runtime/test/transport/messages.test.ts:284]: Have you tried if a shorter time also works? That would be great.

File: packages/consumption/test/modules/requests/OutgoingRequestsController.test.ts
Comment (Milena-Czierlinski) [inline packages/consumption/test/modules/requests/OutgoingRequestsController.test.ts:749]:
```suggestion
                status: LocalRequestStatus.Draft,
```
If the Request was already expired, there'd be no reason for this test.

File: packages/consumption/test/modules/requests/OutgoingRequestsController.test.ts
Comment (Milena-Czierlinski) [inline packages/consumption/test/modules/requests/OutgoingRequestsController.test.ts:710]: Can this be even shorter? If so, please also adjust the sleeps accordingly.

Same for other tests.

File: packages/consumption/test/modules/requests/OutgoingRequestsController.test.ts
Comment (jkoenig134) [inline packages/consumption/test/modules/requests/OutgoingRequestsController.test.ts:1]: this test now runs 8 seconds longer with all those sleeps, this sums up quickly - we have to find an alternative.

Comment (sebbi08) [reply packages/consumption/test/modules/requests/OutgoingRequestsController.test.ts:1]: @Milena-Czierlinski @jkoenig134 I now changed it to 100ms and also added a sleep for 100ms because with lower timing it started to be flaky even locally and also the wait was needed because sometimes the tests were faster.

Comment (sebbi08) [reply packages/consumption/test/modules/requests/OutgoingRequestsController.test.ts:1]: Ok, not working in the CI, so I might need to think about another solution

Comment (Milena-Czierlinski) [reply packages/consumption/test/modules/requests/OutgoingRequestsController.test.ts:1]: Okay, this is already much better. Please apply the same times to the `messages.test`.

Comment (Milena-Czierlinski) [reply packages/consumption/test/modules/requests/OutgoingRequestsController.test.ts:1]: > Ok, not working in the CI, so I might need to think about another solution

That only refers to one test. Maybe it is already sufficient to wait for 200ms there.

Comment (jkoenig134) [reply packages/consumption/test/modules/requests/OutgoingRequestsController.test.ts:1]: > 100ms and also added a sleep for 100ms

wtb 100 and 120?

File: packages/runtime/test/transport/messages.test.ts
Comment (Milena-Czierlinski) [inline packages/runtime/test/transport/messages.test.ts:296]: Please add empty lines around this line.

File: packages/runtime/test/transport/messages.test.ts
Comment (Milena-Czierlinski) [inline packages/runtime/test/transport/messages.test.ts:312]: Empty line below as in the previous test.

File: packages/runtime/test/transport/messages.test.ts
Comment (Milena-Czierlinski) [inline packages/runtime/test/transport/messages.test.ts:322]: Empty line below please.

**Review by Milena-Czierlinski — COMMENTED** [review]
Apart from this I'm fine with the PR.

**Review by Milena-Czierlinski — DISMISSED** [review]

**Review by jkoenig134 — DISMISSED** [review]
LGTM, please wait for another review by @Milena-Czierlinski or @britsta

File: packages/consumption/test/modules/requests/OutgoingRequestsController.test.ts
Comment (britsta) [inline packages/consumption/test/modules/requests/OutgoingRequestsController.test.ts:284]: Please put this test before the `"returns a validation result that contains an error for requests that would lead to the creation of more than one RelationshipAttribute with the same key"` test right above as the Request expiry test is less specific and fits better with the above tests than the Relationship `key` uniqueness test. :D

File: packages/runtime/test/transport/messages.test.ts
Comment (britsta) [inline packages/runtime/test/transport/messages.test.ts:278]: Please adjust the test name to `"should throw correct error for trying to send a Message with an expired Request"`.

("should throw correct error" instead of "should throw correct error message" for consistency with test names above and "an expired Request" instead of "a Request that expired" as it is grammatically preferable.)

File: packages/runtime/test/transport/messages.test.ts
Comment (britsta) [inline packages/runtime/test/transport/messages.test.ts:283]: There isn't a need to define this `requestItem` again as it is already defined within the `beforeAll` block of the tests.

File: packages/runtime/test/transport/messages.test.ts
Comment (britsta) [inline packages/runtime/test/transport/messages.test.ts:314]: This `requestItem` can be omitted as well as it is already defined within the `beforeAll` block.

**Review by britsta — COMMENTED** [review]
Just a few small things, rest LAwesomeTM.

**Review by britsta — APPROVED** [review]

**Pull request merged**
